### PR TITLE
sessionctx: fix tidb_gogc_tuner_threshold 

### PR DESCRIPTION
**Pull request by hawkingrei**
…it is disabled

Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

if ServerMemoryLimit is 0, we need to read machine memory.

Issue Number: close None

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -748,10 +748,16 @@ var defaultSysVars = []*SysVar{
 			}
 			return strconv.FormatFloat(floatValue, 'f', -1, 64), nil
 		},
-		SetGlobal: func(_ context.Context, s *SessionVars, val string) error {
+		SetGlobal: func(_ context.Context, s *SessionVars, val string) (err error) {
 			factor := tidbOptFloat64(val, DefTiDBGOGCTunerThreshold)
 			GOGCTunerThreshold.Store(factor)
 			memTotal := memory.ServerMemoryLimit.Load()
+			if memTotal == 0 {
```
Comment (wshwsh12) [inline sessionctx/variable/sysvar.go:755]:
```suggestion
			if memTotal == 0 && factor > 0 {
```

Comment (hawkingrei) [reply sessionctx/variable/sysvar.go:755]: I think it should be like this.

```
                        if memTotal == 0 {
				memTotal, err = memory.MemTotal()
				if err != nil {
					return err
				}
			}
			if factor > 0 {
				threshold := float64(memTotal) * factor
				gctuner.Tuning(uint64(threshold))
			}
```

**Review by wshwsh12 — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**